### PR TITLE
Refactor AbstractPowerUpFactory to be an interface <organisation

### PR DESCRIPTION
**Pull request by jochem725**
I've changed the abstract class to an interface to be more
consistent with the abstract factory pattern.

### CODE REVIEW

**Comment by bryanvanwijk** [comment]
Seems fine!

File: src/main/java/spaceinvaders/group_22/unit/LifePowerupFactory.java [hunk 1]
```diff
@@ -1,20 +1,28 @@
...
  */
-public class LifePowerupFactory extends AbstractPowerupFactory {
+public class LifePowerupFactory implements AbstractPowerupFactory {
 
 	/**
 	 * Constructor for a LifePowerupFactory.
 	 */
 	public LifePowerupFactory() {
-		super();
+		Game.getLogger().log(getClass().getName() + "  created succesfully", LogEvent.Type.INFO);
```
Comment (doriandekoning) [inline src/main/java/spaceinvaders/group_22/unit/LifePowerupFactory.java:17]: Great way to log! I'll keep that in mind for my code.

**Comment by doriandekoning** [comment]
Looks great!

**Comment by egedebruin** [comment]
Looks good

**Comment by jochem725** [comment]
I'll merge it then.

**Pull request merged**